homeassistant/components/growatt_server/sensor.py

```diff
@@ -588,17 +579,17 @@ def _setup_entity(hass, config, async_add_entities):
     api = growattServer.GrowattApi()
 
     # Log in to api and fetch first plant if no plant id is defined.
-    login_response = api.login(username, password)
+    login_response = await hass.async_add_executor_job(api.login, username, password)
```

Comment:
Please group all of the sync work in a single function that we can schedule once on the executor thread pool.